src/app/pages/system/view-enclosure/stores/enclosure-store.service.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { createServiceFactory, SpectatorService } from '@ngneat/spectator';
+import { mockProvider } from '@ngneat/spectator/jest';
+import { Subject } from 'rxjs';
+import { TestScheduler } from 'rxjs/testing';
+import { getTestScheduler } from 'app/core/testing/utils/get-test-scheduler.utils';
+import {
+  EnclosureDispersalStrategy,
+  MockStorageGenerator,
+  MockStorageScenario,
+} from 'app/core/testing/utils/mock-storage-generator.utils';
+import { TopologyItemType } from 'app/enums/v-dev-type.enum';
+import { ApiEvent } from 'app/interfaces/api-message.interface';
+import {
+  Enclosure,
+  EnclosureElement,
+  EnclosureSlot,
+  EnclosureView,
+} from 'app/interfaces/enclosure.interface';
+import { Pool } from 'app/interfaces/pool.interface';
+import { Disk } from 'app/interfaces/storage.interface';
+import { DialogService, StorageService, WebSocketService } from 'app/services';
+import { EnclosureState, EnclosureStore } from './enclosure-store.service';
+
+describe('EnclosureStore', () => {
+  const websocketSubscription$ = new Subject<ApiEvent<Pool>>();
+  let spectator: SpectatorService<EnclosureStore>;
+  let testScheduler: TestScheduler;
+  const createService = createServiceFactory({
+    service: EnclosureStore,
+    providers: [
+      StorageService,
+      mockProvider(WebSocketService, {
+        subscribe: jest.fn(() => websocketSubscription$),
+      }),
+      mockProvider(DialogService),
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createService();
+    testScheduler = getTestScheduler();
```

Comment:
You only need test scheduler if you want to test observables with `expectObservable` or something similar.
Otherwise you can just use `mockWebsocket` with. `mockCall` like in other tests.